docs/logs/extending-the-sdk/MyClassWithRedactionEnumerator.cs

```diff
@@ -0,0 +1,72 @@
+// <copyright file="MyRedactionProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+internal class MyClassWithRedactionEnumerator : IReadOnlyList<KeyValuePair<string, object>>
+{
+    private static readonly Regex Rgx = new("\\(?\\d{3}[\\.|\\)]?\\d{3}[\\.|\\-]\\d{4}");
+    private readonly IReadOnlyList<KeyValuePair<string, object>> myList;
```

Comment:
```suggestion
    private readonly IReadOnlyList<KeyValuePair<string, object>> traits;
```

Reply:
I've updated name of the private field to be state.
Not sure which one is better, trait or state; but both of them definitely are better than myList :nerd_face:.